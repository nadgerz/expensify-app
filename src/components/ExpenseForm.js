import React, { Component } from 'react';

import moment from 'moment';

import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';


export default class ExpenseForm extends Component {
  onDescriptionChange = ( event ) => {
    const description = event.target.value;
    
    this.setState( () => ({ description }) );
  };
  onNoteChange = ( event ) => {
    const note = event.target.value;
    
    this.setState( () => ({ note }) );
  };
  onAmountChange = ( event ) => {
    const amount = event.target.value;
    
    if ( !amount || amount.match( /^\d+(\.\d{0,2})?$/ ) ) {
      this.setState( () => ({ amount }) );
    }
  };
  onDateChange = ( createdAt ) => {
    
    this.setState( () => ({ createdAt }) );
  };
  onDateChange = ( createdAt ) => {
    
    if ( createdAt ) {
      this.setState( () => ({ createdAt }) );
    }
  };
  onFocusChange = ( { focused } ) => {
    
    this.setState( () => ({ calendarFocused: focused }) );
  };
  onSubmit = ( event ) => {
    event.preventDefault();
    
    // console.log( event );
    
    if ( !this.state.description || !this.state.amount ) {
      this.setState( () => ({ error: 'Please provide description and amount' }) );
    }
    else {
      this.setState( () => ({ error: '' }) );
      
      this.props.onSubmit( {
                             description: this.state.description,
                             amount: parseFloat( this.state.amount, 10 ) * 100,
                             createdAt: this.state.createdAt.valueOf(),
                             note: this.state.note
                           } );
    }
  };
  
  constructor( props ) {
    super( props );
    
    this.state = {
      description: props.expense ? props.expense.description : '',
      note: props.expense ? props.expense.note : '',
      amount: '',
      createdAt: moment(),
      calendarFocused: false,
      error: ''
    };
    
  }
  
  render() {
    return (
      <div>
        
        {
          this.state.error && <p>ERROR: {this.state.error}</p>
        }
        
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            value={this.state.description}
            autoFocus
            onChange={this.onDescriptionChange}
          />
          
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          
          <textarea
            placeholder={'Add a note for your expense (optional)'}
            value={this.state.note}
            onChange={this.onNoteChange}
          >
          </textarea>
          
          <button>Add Expense</button>
        </form>
      
      </div>
    );
  }
}
