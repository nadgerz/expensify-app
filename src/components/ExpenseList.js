import React from 'react';

import { connect } from 'react-redux';


const ExpenseList = () => (
  <div>
    <h1>Expense List</h1>
  </div>
);

const ConnectedExpenseList = connect()( ExpenseList );

export default ExpenseList;