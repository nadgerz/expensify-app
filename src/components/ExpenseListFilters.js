import React, { Component } from 'react';
import { connect } from 'react-redux';

import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

import { setEndDate, setStartDate, setTextFilter, sortByAmount, sortByDate } from '../actions/filters';


class ExpenseListFilters extends Component {
  
  state = {
    calendarFocused: null
  };
  
  onDatesChange = ( { startDate, endDate } ) => {
    console.log( 'onDatesChange' );
    console.log( startDate );
    console.log( endDate );
    
    this.props.dispatch( setStartDate( { startDate } ) );
    this.props.dispatch( setEndDate( { endDate } ) );
  };
  
  onFocusChange = ( { focused } ) => {
    this.setState( () => ({ calendarFocused: focused }) );
  };
  
  render() {
    return (
      <div>
        
        <input
          type="text"
          value={this.props.filters.text}
          onChange={( event ) => {
            // console.log( event.target.value );
            this.props.dispatch( setTextFilter( event.target.value ) );
          }}
        />
        
        <select
          value={this.props.filters.sortBy}
          onChange={( event ) => {
            // console.log( event.target.value );
            
            const selectedValue = event.target.value;
            
            switch ( selectedValue ) {
              case 'amount':
                return this.props.dispatch( sortByAmount() );
              
              case 'date':
                return this.props.dispatch( sortByDate() );
              
              default:
                console.log( 'ERROR: No such option' );
            }
          }}
        >
          <option value="date">Date</option>
          <option value="amount">Amounty</option>
        </select>
        
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId="your_unique_start_date_id"
          
          endDate={this.props.filters.endDate}
          endDateId="your_unique_end_date_id"
          
          onDatesChange={this.onDatesChange}
          
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
          small={true}
        />
      
      </div>
    );
  }
}


const mapStateToProps = ( state ) => {
  return {
    filters: state.filters
  };
};

export default connect( mapStateToProps )( ExpenseListFilters );
