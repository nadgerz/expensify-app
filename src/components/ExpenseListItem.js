import React from 'react';


const ExpenseListItem = ( props ) => (
  <div>
    <p>{props.description}</p>
    <p>{props.amount}</p>
    <p>{props.createdAt}</p>
  </div>
);

export default ExpenseListItem;