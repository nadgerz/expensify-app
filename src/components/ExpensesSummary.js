import React from 'react';

import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';


export const ExpensesSummary = ( props ) => {
  
  const expensStr = props.expenses.length === 1 ? 'expense' : 'expenses';
  console.log( props.expenses );
  
  return (
    <div>
      {expensStr}
    </div>
  );
};

const mapStateToProps = ( state ) => {
  return {
    expenses: selectExpenses( state.expenses, state.filters )
  };
};

export default connect( mapStateToProps )( ExpensesSummary );
