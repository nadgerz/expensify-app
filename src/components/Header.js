import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => (
  <header>
    
    <h1>Expensify</h1>
    
    <NavLink to="/" activeClassName={'is-active'} exact={true}>Dashboard</NavLink>
    <NavLink to="/create" activeClassName={'is-active'}>Create</NavLink>
    <NavLink to="/help" activeClassName={'is-active'}>Help</NavLink>
    {/*<NavLink to="/404" activeClassName={'is-active'}>404</NavLink>*/}
    
    <button>Logout</button>
  
  </header>
);

export default Header;