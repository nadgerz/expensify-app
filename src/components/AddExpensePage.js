import React from 'react';
import { connect } from 'react-redux';

import { addExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';


const AddExpensePage = ( props ) => (
  <div>
    <h1>Add Expense</h1>
    
    <ExpenseForm
      onSubmit={( expense ) => {
        // console.log( 'ADD!' );
        // console.log( expense );
        
        // props.dispatch( addExpense( expense ) );
        props.onSubmit( expense );
        
        props.history.push( '/' );
      }}
    />
  </div>
);

const mapDispatchToProps = ( dispatch ) => {
  return {
    onSubmit: ( expense ) => dispatch( addExpense( expense ) )
  };
};

//
//    Do not need anything from state, so we can have an empty connect()
//
export default connect( undefined, mapDispatchToProps )( AddExpensePage );