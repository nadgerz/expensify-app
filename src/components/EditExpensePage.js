import React from 'react';
import { connect } from 'react-redux';

import { editExpense, removeExpense } from '../actions/expenses';

import ExpenseForm from './ExpenseForm';


const EditExpensePage = ( props ) => {
  console.log( props );
  
  const handleOnClick = ( id ) => {
    props.dispatch( removeExpense( { id } ) );
    
    props.history.push( '/' );
  };
  
  return (
    <div>
      <p>Edit Expense</p>
      
      <ExpenseForm
        expense={props.expense}
        
        onSubmit={( expense ) => {
          // console.log( 'EDIT!' );
          // console.log( expense );
          // console.log( props.match.params.id );
          props.dispatch( editExpense( props.expense.id, expense ) );
          
          props.history.push( '/' );
        }}
      />
      
      <button onClick={handleOnClick( props.match.params.id )}>Remove</button>
    
    </div>
  );
};

const mapStateToProps = ( state, props ) => {
  return {
    expense: state.expenses.find( ( expense ) => {
      return expense.id === props.match.params.id;
    } )
  };
};

export default connect( mapStateToProps )( EditExpensePage );