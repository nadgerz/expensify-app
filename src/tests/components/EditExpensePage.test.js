import React from 'react';
import { shallow } from 'enzyme';

import { EditExpensePage } from '../../components/EditExpensePage';
// import expenses from '../fixtures/expenses';

test( 'should render EditExpensePage correctly', () => {
  const wrapper = shallow( <EditExpensePage/> );
  
  expect( wrapper ).toMatchSnapshot();
} );