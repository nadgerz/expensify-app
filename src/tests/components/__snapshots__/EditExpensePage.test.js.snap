// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage correctly 1`] = `
<div>
  <p>
    Edit Expense
  </p>
  <ExpenseForm
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
